src/textual/css/parse.py

```diff
@@ -143,14 +164,82 @@ def parse_rule_set(
         rule_selectors.append(selectors[:])
 
     declaration = Declaration(token, "")
-
     errors: list[tuple[Token, str | HelpText]] = []
+    nested_rules: list[RuleSet] = []
 
     while True:
         token = next(tokens)
+
         token_name = token.name
         if token_name in ("whitespace", "declaration_end"):
             continue
+        if token_name in {
+            "selector_start_id",
+            "selector_start_class",
+            "selector_start_universal",
+            "selector_start",
+            "nested",
+        }:
+            recursive_parse: list[RuleSet] = list(
+                parse_rule_set(
+                    "",
+                    tokens,
+                    token,
+                    is_default_rules=is_default_rules,
+                    tie_breaker=tie_breaker,
+                )
+            )
+
+            def combine_selectors(
+                selectors1: list[Selector], selectors2: list[Selector]
+            ) -> list[Selector]:
+                """Combine lists of selectors together, processing any nesting.
+
+                Args:
+                    selectors1: List of selectors.
+                    selectors2: Second list of selectors.
+
+                Returns:
+                    Combined selectors.
+                """
+                if selectors2 and selectors2[0].type == SelectorType.NESTED:
+                    final_selector = selectors1[-1]
```

Comment:
Don't we need to protect ourselves against `selectors1` being empty in a degenerate case with malformed CSS?

Reply:
I don't believe that can occur. The parser will ensure that there is at least one selector.